pex/dist_metadata.py

```diff
@@ -101,73 +133,69 @@ def _parse_sdist_package_info(sdist_path):
     return None
 
 
+@attr.s(frozen=True)
+class DistMetadataFile(object):
+    project_name = attr.ib()  # type: ProjectName
+    version = attr.ib()  # type: Version
+    path = attr.ib()  # type: str
+
+
+def find_dist_info_files(
+    filename,  # type: Text
+    listing,  # type: Iterable[str]
+):
+    # type: (...) -> Iterator[DistMetadataFile]
+    dist_info_metadata_pattern = "^{}$".format(
+        os.path.join(r"(?P<project_name>.+)-(?P<version>.+)\.dist-info", re.escape(filename))
```

Comment:
I'll de-normalize some comments here. The situation with version is spelled out in Version in pep_440.py. You _can_ in fact have a `-` and wheel normalizes this to `_` and Pip deals with that as does Pex. This is all only sane because Pex vendors wheel and Pip and is ensured it has versions of both that are stable and handle this case stably.